### PR TITLE
Edge cutoff Prefab - initial 

### PR DESCRIPTION
**Pull request by cozloff**
# Description
The identification and swapping of edge tiles with specialized prefab variants, such as peninsulas and corners, to create a more visually appealing and dynamically structured map. 

## Implemented Features
Edge Tile Detection and Swapping: The core addition in this update is the logic for detecting edge tiles based on their neighboring tiles' positions. Depending on whether an edge tile is identified as a peninsula (North, South, East, West) or a corner (NE, NW, SE, SW), it is dynamically swapped with a corresponding prefab. This swap enhances the visual distinction between regular tiles and those on the map's edge, adding depth and interest to the generated maps.

## Prefab Creation for Edge Tiles: 
Alongside the script update, this pull request includes the creation of new prefab assets for each edge tile variant. These prefabs (humpEastPrefab, humpNorthPrefab, humpSouthPrefab, humpWestPrefab, NEcornerPrefab, NWcornerPrefab, SEcornerPrefab, SWcornerPrefab) are designed to visually represent their respective edge types, with rounded corners for peninsulas and more pronounced corner designs for the corner tiles.

## Isometric Adjustments: 
Recognizing the unique requirements of isometric map generation, adjustments were made to ensure that the detection of neighboring tiles and the swapping mechanism work correctly within an isometric grid. This involved modifying the directional checks (isoUp, isoDown, isoLeft, isoRight) to align with isometric coordinates, ensuring accurate edge tile identification.

## Conclusion
This update represents a significant enhancement to the map generation capabilities of our project, offering a more refined and visually engaging output. By incorporating specialized edge tiles, we introduce variety and aesthetic appeal to the procedurally generated maps, elevating the overall user experience.

### CODE REVIEW

**Comment by tnishamon** [comment]
Hey Dylan, just a heads up that in your fork for this PR, the Unity version you used it 2022.3.22f1, make sure to update your version back to 2022.3.19f1 before we merge it.

File: 2DRove/ProjectSettings/ProjectVersion.txt
Comment (tnishamon) [inline 2DRove/ProjectSettings/ProjectVersion.txt:1]: Make sure that we are on the same Unity version

**Comment by SwiftNemesis** [comment]
Also want to note, please change it to a draft until it's ready to be fully tested during review.

I went ahead and did so for now. Please click "Ready for review" once it's updated so me and @tnishamon can review it.

Looking forward to reviewing further! :)

File: 2DRove/.history/Assets/Scripts/MapGenDLA_20240325125152.cs
Comment (tnishamon) [inline 2DRove/.history/Assets/Scripts/MapGenDLA_20240325125152.cs:1]: This is more of a comment to keep in mind for the future, but we should probably make the attributes in our tiles into a separate class script to reference. This makes testing easier and cleans up our code a little bit :)

**Review by tnishamon — CHANGES_REQUESTED** [review]

**Review by tnishamon — APPROVED** [review]
I like the look of the new implementation of edge tiles. Great job, Dylan! The way borders are generated get a little funky with the edge tiles, but we need to fix that implementation anyways. 

I will work on making sure our border generation works with the new changes.

**Pull request merged**